codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java

```diff
@@ -604,16 +606,15 @@ private void parseBodyAttributes() {
                 // special case
                 ampersandpos = currentpos;
                 if (ampersandpos > firstpos) {
-                    setFinalBuffer(undecodedChunk.retainedSlice(firstpos, ampersandpos - firstpos));
+                    setFinalBuffer(undecodedChunk.copy(firstpos, ampersandpos - firstpos));
```

Comment:
hmm... retainedSlice(...) should be fine. This will do an extra memory copy which I think is not needed.